client/my-sites/plans-grid/components/plan-type-selector/hooks/use-term-vise-max-dicounts.ts

```diff
@@ -0,0 +1,97 @@
+import {
+	PlanSlug,
+	URL_FRIENDLY_TERMS_MAPPING,
+	UrlFriendlyTermType,
+	getPlanMultipleTermsVariantSlugs,
+	getPlanSlugForTermVariant,
+	isMonthly,
+	isWpComPlan,
+	isWpcomEnterpriseGridPlan,
+} from '@automattic/calypso-products';
+import { UsePricingMetaForGridPlans } from 'calypso/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans';
+
+export default function useTermViseMaxDiscounts(
```

Comment:
`NB:`

Would you mind explaining a bit more about the name "TermVise"? I'm not familiar with the concept

Reply:
Let's rename the hook to `useMaxDiscountsForPlanTerms`. It seems like it better describes both the keys and values of the mapping returned by this hook. p1702325681579499/1702307941.915969-slack-CV2TX2PAN

Reply:
> So first what i do is convert the term price to monthly terms.

OH! This is the step I was trying to wrap my head around. Do you mind adding an inline comment explaining this for now?